---
layout: default
title: Publications
---

## Preprints
**Sparse-Input Neural Networks for High-dimensional Nonparametric Regression and Classification**<br />
*Feng* and Simon<br />
[\[arXiv\]](http://arxiv.org/abs/1711.07592)[\[code\]](https://github.com/jjfeng/spinn)

## 2022
**Bayesian logistic regression for online recalibration and revision of risk prediction models with performance guarantees**<br />
*Feng*, Gossmann, Sahiner and Pirracchio<br />
*Journal of the American Medical Informatics Association*, 2022<br />
[\[paper\]](https://doi.org/10.1093/jamia/ocab280)[\[code\]](https://github.com/jjfeng/bayesian_model_revision)

**Ensembled sparse-input hierarchical networks for high-dimensional datasets**<br />
*Feng* and Simon<br />
*Statistical Analysis and Data Mining*, 2022<br />
[\[paper\]](http://doi.org/10.1002/sam.11579)[\[code\]](https://pypi.org/project/EASIER-net/)

**Overcoming barriers in the design and implementation of clinical trials for Acute Kidney Injury: a report from the 2020 Kidney Disease Clinical Trialists meeting**<br />
Lazzareschi, Mehta, Dember, Bernholz, Turan, Sharma, Kheterpal, Parikh, Ali, Schulman, Ryan, *Feng*, Simon, Pirracchio, Rossignol and Legrand<br />
*Nephrol. Dial. Transplant*, 2022<br />
[\[paper\]](http://dx.doi.org/10.1093/ndt/gfac003)

**Clinical artificial intelligence quality improvement: towards continual monitoring and updating of AI algorithms in healthcare**<br />
*Feng*, Phillips, Malenica, Bishara, Hubbard, Celi and Pirracchio<br />
*npj Digital Medicine*, 2022<br />
[\[paper\]](https://www.nature.com/articles/s41746-022-00611-y)

**Prostate cancer therapy personalization via multi-modal deep learning on randomized phase III clinical trials**<br />
Esteva, *Feng*, van der Wal, Huang, Simko, DeVries, Chen, Schaeffer, Morgan, Sun, Ghorbani, Naik, Nathawani, Socher, Michalski, Roach, Pisansky, Monson, Naz, Wallace, Ferguson, Bahary, Zou, Lungren, Yeung, Ross, Sandler, Tran, Spratt, Pugh, Feng and Mohamad<br />
*npj Digital Medicine*, 2022<br />
[\[paper\]](https://www.nature.com/articles/s41746-022-00613-w)

**Sequential algorithmic modification with test data reuse**<br />
*Feng*, Pennello, Petrick, Sahiner, Pirracchio and Gossmann<br />
*Proceedings of the Thirty-Seventh Conference on Uncertainty in Artificial Intelligence*, 2022<br />
[\[paper\]](https://openreview.net/forum?id=rKb8gLLs5g5)[\[code\]](https://github.com/jjfeng/adaptive_SRGP)[\[poster\]](adaptive_SRGP.pdf)

## 2021
**Approval policies for modifications to Machine Learning-Based Software as a Medical Device: A study of bio-creep**<br />
*Feng*, Emerson and Simon<br />
*Biometrics*, 2021<br />
[\[paper\]](https://doi.org/10.1111/biom.13379)[\[code\]](https://github.com/jjfeng/aACP)

**Estimation of cell lineage trees by maximum-likelihood phylogenetics**<br />
*Feng*, DeWitt, McKenna, Simon, Willis and Matsen<br />
*Annals of Applied Statistics*, 2021<br />
[\[paper\]](https://projecteuclid.org/journals/annals-of-applied-statistics/volume-15/issue-1/Estimation-of-cell-lineage-trees-by-maximum-likelihood-phylogenetics/10.1214/20-AOAS1400.short)[\[code\]](https://github.com/matsengrp/gapml)

**Learning to safely approve updates to machine learning algorithms**<br />
*Feng*<br />
*Proceedings of the Conference on Health, Inference, and Learning*, 2021<br />
[\[paper\]](https://doi.org/10.1145/3450439.3451864)[\[code\]](https://github.com/jjfeng/aACP_time_trends)

**Selective prediction-set models with coverage guarantees**<br />
*Feng*, Sondhi, Perry and Simon<br />
*Biometrics*, 2021<br />
[\[paper\]](http://doi.org/10.1111/biom.13612)[\[code\]](https://github.com/jjfeng/pc_SPS)

## 2020
**Efficient nonparametric statistical inference on population feature importance using Shapley values**<br />
Williamson and *Feng*<br />
*International Conference on Machine Learning (ICML)*, 2020<br />
[\[paper\]](http://proceedings.mlr.press/v119/williamson20a.html)[\[code\]](https://github.com/bdwilliamson/spvim_supplementary)

**An analysis of the cost of hyper-parameter selection via split-sample validation, with applications to penalized regression**<br />
*Feng* and Simon<br />
*Statistica Sinica*, 2020<br />
[\[paper\]](http://www3.stat.sinica.edu.tw/ss_newpaper/SS-2017-0310_na.pdf)

## 2019
**Deep generative models for T cell receptor protein sequences**<br />
Davidsen, Olson, DeWitt, *Feng*, Harkins, Bradley and Matsen<br />
*Elife*, 2019<br />
[\[paper\]](http://dx.doi.org/10.7554/eLife.46935)[\[code\]](https://github.com/matsengrp/vampire/)

**Survival analysis of DNA mutation motifs with penalized proportional hazards**<br />
*Feng*, Shaw, Minin, Simon and Matsen<br />
*Ann. Appl. Stat.*, 2019<br />
[\[paper\]](https://projecteuclid.org/euclid.aoas/1560758446)[\[code\]](https://github.com/matsengrp/samm)

## 2018
**Gradient-based Regularization Parameter Selection for Problems With Nonsmooth Penalty Functions**<br />
*Feng* and Simon<br />
*J. Comput. Graph. Stat.*, 2018<br />
[\[paper\]](https://doi.org/10.1080/10618600.2017.1390470)[\[code\]](https://github.com/jjfeng/nonsmooth-joint-opt)

**Nonparametric variable importance using an augmented neural network with multi-task learning**<br />
*Feng*, Williamson, Simon and Carone<br />
*International Conference on Machine Learning (ICML)*, 2018<br />
[\[paper\]](http://proceedings.mlr.press/v80/feng18a.html)[\[code\]](https://github.com/jjfeng/nnet_var_import)